### PR TITLE
Enable sidebar register function to get the sidebars in sidebar container across all the app

### PR DESCRIPTION
**Pull request by Sifawm**
New function inside SidebarService to registry sidebar with observables.

### CODE REVIEW

File: src/sidebar.component.ts [hunk 1]
```diff
@@ -76,8 +75,7 @@ import { isBrowser, upperCaseFirst, isLTR, isIOS } from './utils';
       transition: transform 0.3s cubic-bezier(0, 0, 0.3, 1);
     }
   `],
-  changeDetection: ChangeDetectionStrategy.OnPush,
-  encapsulation: ViewEncapsulation.None
```
Comment (arkon) [inline src/sidebar.component.ts]: Why was this removed?

File: src/sidebar.service.ts [hunk 2]
```diff
@@ -15,11 +19,19 @@ export class SidebarService {
     this._closeObserver.next();
   }
 
+  registry(sidebar: Sidebar) {
```
Comment (arkon) [inline src/sidebar.service.ts]: Maybe call this `register`?

File: src/sidebar-container.component.ts [hunk 3]
```diff
@@ -260,7 +248,7 @@ export class SidebarContainer implements AfterContentInit, OnChanges, OnDestroy
     if (this._sidebars && this.allowSidebarBackdropControl) {
       let hasOpen = false;
 
-      const _sidebars = this._sidebars.toArray();
+      const _sidebars: Sidebar[] = [...this._sidebars];
```
Comment (arkon) [inline src/sidebar-container.component.ts]: Is this necessary?

File: src/sidebar-container.component.ts [hunk 4]
```diff
@@ -87,22 +82,21 @@ export class SidebarContainer implements AfterContentInit, OnChanges, OnDestroy
   @Output() showBackdropChange = new EventEmitter<boolean>();
 
   /** @internal */
-  @ContentChildren(Sidebar)
-  _sidebars: QueryList<Sidebar>;
+  _sidebars: Sidebar[] =  new Array<Sidebar>();
```
Comment (arkon) [inline src/sidebar-container.component.ts]: I prefer just `[]` over the `new Array()` style.

File: demo/src/demo/demo.component.ts [hunk 5]
```diff
@@ -3,90 +3,88 @@ import { Component, AnimationTransitionEvent } from '@angular/core';
...
+
+      <p>Text from <a href="http://www.catipsum.com/">Cat Ipsum</a>.</p>
+    </section>
   `
 })
 export class DemoComponent {
   private _opened: boolean = false;
-  private _modeNum: number = 0;
+  private _modeNum: number = 1;
   private _positionNum: number = 0;
```
Comment (arkon) [inline demo/src/demo/demo.component.ts]: Why?

File: src/sidebar.service.ts (hunk 2 above)
Comment (Sifawm) [inline src/sidebar.service.ts]: I think that you have reason. Register is better word for the function.

File: src/sidebar.component.ts (hunk 1 above)
Comment (Sifawm) [inline src/sidebar.component.ts]: I only removed ViewEncapsulation.None. In my opinion, I prefer to use the default value if it is possible (emulated) as a solution to the future of browsers with shadow sun. Do you think that is better ViewEncapulation.None or Emulated?

File: src/sidebar-container.component.ts (hunk 3 above)
Comment (Sifawm) [inline src/sidebar-container.component.ts]: Posibibly not. We can use directly 
`
const _sidebars = this._sidebars;
`

or use the property of the class directly as for argument.

Comment (arkon) [inline src/sidebar-container.component.ts]: Yeah I think we can just directly use it if it's already an array.

File: src/sidebar.component.ts (hunk 1 above)
Comment (arkon) [inline src/sidebar.component.ts]: I suppose it's fine to keep it as the default. Not entirely sure if that'll affect users though.

Comment (Sifawm) [inline src/sidebar.component.ts]: The users needs to use the deep (>>>) on css to modify the style of the sidebar if the css file is loaded by angular. On other cases, the deep selector is not necessary.

**Comment by Sifawm** [comment]
@arkon all changes done

**Comment by Sifawm** [comment]
@arkon do you see any changes more?

**Comment by arkon** [comment]
@Sifawm Yup, I'm on vacation right now though so I don't have access to a laptop to really do much, so I'll merge/publish when I get back home at the end of the week.

**Comment by Sifawm** [comment]
@arkon my apologises. Enjoy your vacations.
